src/server/req_cred.c

```diff
@@ -184,7 +184,7 @@ get_cached_cred(job  *pjob)
 	}
 
 	strncpy(cred->credid, pjob->ji_wattr[(int)JOB_ATR_cred_id].at_val.at_str,
-		strlen(pjob->ji_wattr[(int)JOB_ATR_cred_id].at_val.at_str));
+		strlen(pjob->ji_wattr[(int)JOB_ATR_cred_id].at_val.at_str) + 1); /* +1 for '\0' */
```

Comment:
This use of strncpy does not seem to be any more safe that a direct strcpy. This intends to copy ALL the bytes from the source string to the destination irrespective of the size of the destination buffer. strncpy is useful only if we use the destination size in the 3rd argument in an effort to not cross of the destination length, and in that case we always explicitly  put a null character at the last position of the destination string.

Reply:
Thanks, it makes sense. Changed.